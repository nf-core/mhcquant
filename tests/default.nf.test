nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("-profile test") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // Extract peptidoform column from all TSV files
            def peptidoform_data = []
            new File(params.outdir).eachFileRecurse { file ->
                if (file.name.endsWith('.tsv')) {
                    def lines = file.readLines()
                    if (lines.size() > 0) {
                        def header = lines[0].split('\t')
                        def peptidoformIndex = header.findIndexOf { it == 'peptidoform' }
                        if (peptidoformIndex >= 0) {
                            def peptidoforms = lines.drop(1).collect { line ->
                                def fields = line.split('\t')
                                fields.size() > peptidoformIndex ? fields[peptidoformIndex] : ''
                            }.findAll { it != '' }.sort()
                            peptidoform_data.add([file.name, peptidoforms])
                        }
                    }
                }
            }
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mhcquant_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // Peptidoform data from TSV files
                    peptidoform_data
                ).match() }
            )
        }
    }
}
