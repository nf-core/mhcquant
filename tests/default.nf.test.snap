{
    "-profile test": {
        "content": [
            21,
            {
                "MS2RESCORE": {
                    "MS\u00b2Rescore": "3.1.5)"
                },
                "OPENMSTHIRDPARTY_COMETADAPTER": {
                    "CometAdapter": "3.4.0",
                    "Comet": "2024.01 rev. 1"
                },
                "OPENMS_DECOYDATABASE": {
                    "openms": "3.4.0"
                },
                "OPENMS_IDFILTER_Q_VALUE": {
                    "openms": "3.4.0"
                },
                "OPENMS_IDMASSACCURACY": {
                    "OpenMS": "3.4.0"
                },
                "OPENMS_IDMERGER": {
                    "openms": "3.4.0"
                },
                "OPENMS_PEPTIDEINDEXER": {
                    "openms": "3.4.0"
                },
                "OPENMS_PERCOLATORADAPTER": {
                    "PercolatorAdapter": "3.4.0-pre-exported-20250510",
                    "percolator": "3.07.1, Build Date Mar 13 2025 17:19:27"
                },
                "OPENMS_PSMFEATUREEXTRACTOR": {
                    "openms": "3.4.0"
                },
                "OPENMS_TEXTEXPORTER": {
                    "openms": "3.4.0"
                },
                "PYOPENMS_CHROMATOGRAMEXTRACTOR": {
                    "python": "3.13.1",
                    "pyopenms": "3.3.0"
                },
                "SUMMARIZE_RESULTS": {
                    "python": "3.13.1",
                    "pyopenms": "3.3.0"
                },
                "Workflow": {
                    "nf-core/mhcquant": "v3.1.0dev"
                }
            },
            [
                "HepG2_A.tsv",
                "intermediate_results",
                "intermediate_results/comet",
                "intermediate_results/comet/HepG2_rep1_small_pin.tsv",
                "intermediate_results/comet/HepG2_rep2_small_pin.tsv",
                "intermediate_results/comet/HepG2_rep3_small_pin.tsv",
                "intermediate_results/rescoring",
                "intermediate_results/rescoring/HepG2_A_ms2rescore.idXML",
                "intermediate_results/rescoring/HepG2_A_pout.idXML",
                "intermediate_results/rescoring/HepG2_A_pout_filtered.idXML",
                "intermediate_results/rescoring/HepG2_A_psm.idXML",
                "multiqc",
                "multiqc/ms2rescore",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_chromatogram.txt",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_histogram_mz.txt",
                "multiqc/multiqc_data/multiqc_histogram_rt.txt",
                "multiqc/multiqc_data/multiqc_histogram_scores.txt",
                "multiqc/multiqc_data/multiqc_length_dist.txt",
                "multiqc/multiqc_data/multiqc_mass_error.txt",
                "multiqc/multiqc_data/multiqc_scores_xcorr.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/chromatogram-cnt.pdf",
                "multiqc/multiqc_plots/pdf/chromatogram-log.pdf",
                "multiqc/multiqc_plots/pdf/general_stats.pdf",
                "multiqc/multiqc_plots/pdf/histogram_mz.pdf",
                "multiqc/multiqc_plots/pdf/histogram_rt.pdf",
                "multiqc/multiqc_plots/pdf/histogram_scores.pdf",
                "multiqc/multiqc_plots/pdf/length_dist.pdf",
                "multiqc/multiqc_plots/pdf/mass_error.pdf",
                "multiqc/multiqc_plots/pdf/scores_xcorr.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/chromatogram-cnt.png",
                "multiqc/multiqc_plots/png/chromatogram-log.png",
                "multiqc/multiqc_plots/png/general_stats.png",
                "multiqc/multiqc_plots/png/histogram_mz.png",
                "multiqc/multiqc_plots/png/histogram_rt.png",
                "multiqc/multiqc_plots/png/histogram_scores.png",
                "multiqc/multiqc_plots/png/length_dist.png",
                "multiqc/multiqc_plots/png/mass_error.png",
                "multiqc/multiqc_plots/png/scores_xcorr.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/chromatogram-cnt.svg",
                "multiqc/multiqc_plots/svg/chromatogram-log.svg",
                "multiqc/multiqc_plots/svg/general_stats.svg",
                "multiqc/multiqc_plots/svg/histogram_mz.svg",
                "multiqc/multiqc_plots/svg/histogram_rt.svg",
                "multiqc/multiqc_plots/svg/histogram_scores.svg",
                "multiqc/multiqc_plots/svg/length_dist.svg",
                "multiqc/multiqc_plots/svg/mass_error.svg",
                "multiqc/multiqc_plots/svg/scores_xcorr.svg",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_mhcquant_software_mqc_versions.yml"
            ],
            [
                "multiqc_chromatogram.txt:md5,f2f618e6fea89293377cd23ea3b37367",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "multiqc_general_stats.txt:md5,804fbfbdf38330e7baf4df7620f9397c",
                "multiqc_histogram_mz.txt:md5,9dcadc47d65c40deca3dc8aeb1d8bcb9",
                "multiqc_histogram_rt.txt:md5,4f009097be107a686cd45959f8e12e94",
                "multiqc_histogram_scores.txt:md5,430ddfa4200703a46f9cdb387c84220c",
                "multiqc_mass_error.txt:md5,68f907e91534decaa47c2acf49d4c2ef",
                "multiqc_scores_xcorr.txt:md5,f4eaa7c889e5a4eb912fb7a9bb1e700c"
            ],
            [
                [
                    "HepG2_A.tsv",
                    [
                        "AAAAAAHRL",
                        "AAATVTKKV",
                        "AAHPNVQKV",
                        "AAKDPSPSV",
                        "AAM(Oxidation)SVAKRV",
                        "AANAQPHKL",
                        "AANDKLKKM",
                        "AANPDPKKL",
                        "AAPPPPAHA",
                        "AAPRPPPKPM(Oxidation)",
                        "AAQAHTREL",
                        "AAQSEAKVV",
                        "AARAEAREI",
                        "AASPVGKRL",
                        "AASSLKKQY",
                        "AATDHSNQL",
                        "AATPAKKTV",
                        "AATRWAKKI",
                        "AATSNLKKL",
                        "AATVTKKTY",
                        "AKDVKFGAD",
                        "ALKDKIEKA",
                        "ALKDSVQRA",
                        "ALKENGGARLA",
                        "ALSDLHAHK",
                        "AM(Oxidation)AARPHSI",
                        "AM(Oxidation)AASPHAV",
                        "AM(Oxidation)ASHLTST",
                        "AM(Oxidation)HGHVEVV",
                        "AM(Oxidation)NARPHKV",
                        "AMNARPHKV",
                        "APRPPPKPM(Oxidation)",
                        "ATWSDPHKA",
                        "AVM(Oxidation)GNPKVKA",
                        "AVVVKKIETRDG",
                        "DAAVHQRI",
                        "DAKKQIKQV",
                        "DAPAGRKV",
                        "DGKKKISTV",
                        "DHVITNM(Oxidation)N",
                        "DPM(Oxidation)KARVV",
                        "DPPKTHVT",
                        "DPPRDSKGL",
                        "DSKKSQPKI",
                        "EAKTHEAQI",
                        "EDRDDNKVY",
                        "EYSKQM(Oxidation)QRF",
                        "FADPHSKRV",
                        "FPHSAHQKY",
                        "FPSRGKSSSY",
                        "GKVGAHAGEY",
                        "GKVGAHAGEYG",
                        "GLAKSIHHA",
                        "GMAKKVTKA",
                        "GNVAEGETKPD",
                        "GPNKKPRF",
                        "GRPM(Oxidation)PPSRR",
                        "GRVVEPKR",
                        "GVM(Oxidation)VGM(Oxidation)GQKD",
                        "HAAENPGKY",
                        "HAANPNGRYY",
                        "HAATYTKKV",
                        "HAFGGNKASY",
                        "HAGPTAHEL",
                        "HAIGKSQSF",
                        "HAKEKLTAL",
                        "HAM(Oxidation)EEVKKF",
                        "HAPELTKKV",
                        "HLKEDQTEY",
                        "HPAENGKSNF",
                        "HPESERISM(Oxidation)",
                        "HPNKVKVY",
                        "HPSQPQGTY",
                        "HPYQPDPKM(Oxidation)",
                        "HSRNEGVATY",
                        "HSSPASSNY",
                        "HVKDANGNSF",
                        "HVSTHQHNY",
                        "IASQTVKKV",
                        "IKTVETRD",
                        "ILDKKVEKV",
                        "IPKKHISQI",
                        "IPQKFHRSV",
                        "ISSEAHREV",
                        "IVGRPRHQ",
                        "IYTEHAHQV",
                        "KAAAFEKQV",
                        "KAAKPKVV",
                        "KAASHIKSI",
                        "KAATVTKEL",
                        "KAISGVHTV",
                        "KAKPGTYKV",
                        "KALVKPKEV",
                        "KASEQAKVV",
                        "KASGTLREY",
                        "KAYGKAKSI",
                        "KAYSEAHEI",
                        "KHANQVLSL",
                        "KLAESVEKA",
                        "KLAPPPKKA",
                        "KLDNQVSKV",
                        "KLM(Oxidation)QQQQEV",
                        "KLQPGSVKKV",
                        "KLSPKAEEV",
                        "KM(Oxidation)DESKHEI",
                        "KM(Oxidation)LDHEYTT",
                        "KMDESKHEI",
                        "KMQEHSDQV",
                        "KSKEFVQKV",
                        "KTKEGVREV",
                        "KVAQPKEV",
                        "KVIDTQQKV",
                        "KVSNSGITRV",
                        "KVVNPTQK",
                        "KYLGKTHNL",
                        "KYNRQSM(Oxidation)TL",
                        "KYQPKPKQI",
                        "KYTPPPHHI",
                        "LAKTGVHHY",
                        "LDKKVEKV",
                        "LLKSEKSSY",
                        "LPSENHKEM",
                        "LPSENHKEM(Oxidation)",
                        "M(Oxidation)LSEHTSKL",
                        "M(Oxidation)PPEHVRKI",
                        "M(Oxidation)PQGGGQHY",
                        "M(Oxidation)PQKERESI",
                        "M(Oxidation)PRNLREY",
                        "MPKAEKNAV",
                        "MPNKVRKI",
                        "MPPEHVRKI",
                        "MPQKERESI",
                        "NAAEGRSF",
                        "NAHGEPRNY",
                        "NALAHKYH",
                        "NAM(Oxidation)KIRDV",
                        "NASARDNTI",
                        "NATAVVRHM(Oxidation)",
                        "NATNAVKRL",
                        "NGTHVVRHY",
                        "NIQKITKS",
                        "NLAEKPKTV",
                        "NLKVKGPVR",
                        "NLRPKKKV",
                        "NLRPKKKVK",
                        "NLTEEEEKSKS",
                        "NNLSSLSKK",
                        "NPAAYENDK",
                        "NPFGGASHAKG",
                        "NPKESSSSL",
                        "NTKIGPRR",
                        "NVDYSKLKK",
                        "NVINGGSHA",
                        "NVKIVKVKKE",
                        "NYTKEHNNL",
                        "PKSKELVSSSS",
                        "PPNKKPKV",
                        "PVEKALRD",
                        "QAGPINREM(Oxidation)",
                        "QALKHLKQY",
                        "QGVM(Oxidation)VGM(Oxidation)GQK",
                        "QGVM(Oxidation)VGM(Oxidation)GQKD",
                        "QPAKADKESF",
                        "QPHKVTQY",
                        "QTHQPPAPNS",
                        "QVKPNSNQV",
                        "QYKKDGADF",
                        "RADKKAPR",
                        "RLAEAHAKV",
                        "RLHEKIQEA",
                        "RLNNKSAKV",
                        "RPKKPGQSF",
                        "RPVPKGATY",
                        "RQVTGVTRV",
                        "RSNEHIREL",
                        "RSTENVNKL",
                        "RSTQYM(Oxidation)KEY",
                        "RVAGIHKKV",
                        "RYDDM(Oxidation)AAAM(Oxidation)",
                        "RYIDLHSNR",
                        "RYIDTHNRV",
                        "RYM(Oxidation)NHM(Oxidation)QSL",
                        "SAKLERSHY",
                        "SAM(Oxidation)SNPRAM",
                        "SAQTTSHEL",
                        "SARAGETRF",
                        "SASEAAKKY",
                        "SASHVSKTV",
                        "SATGHPRKV",
                        "SATNALRSM(Oxidation)",
                        "SATSNKHLL",
                        "SAVDRQREY",
                        "SAYANAKKY",
                        "SEVAHRFK",
                        "SIVGRPRH",
                        "SKEYFSKQ",
                        "SLDKTSHSV",
                        "SLDTQPKKV",
                        "SM(Oxidation)LGSPHKV",
                        "SM(Oxidation)NTHLKAV",
                        "SPM(Oxidation)EKIKQY",
                        "SPNYHEEKY",
                        "SQRFPKAE",
                        "SVKNDHSAY",
                        "SVQVFGRKK",
                        "SYVGDEAQSKR",
                        "SYVGDEAQSKRG",
                        "TALAPSTM(Oxidation)K",
                        "TAPQTQHV",
                        "TARPGPRAV",
                        "TDKSFVEK",
                        "TDRETGKL",
                        "TEKLVTSK",
                        "TEKLVTSKG",
                        "TGYYGKGKP",
                        "THQPPAPNS",
                        "THYDPPRK",
                        "TIGEGQQHHLG",
                        "TISRPKGVA",
                        "TKELPSGKKY",
                        "TLVQTKGT",
                        "TNNLRPKKKVK",
                        "TPPGVRKI",
                        "TRLSRTPGNR",
                        "TSGPGSRISSSS",
                        "TSTPNAKTV",
                        "TTPISEQKG",
                        "TTTHITKTV",
                        "TVIGPDGHKE",
                        "TVKDSRTVY",
                        "TVQNEANKY",
                        "VAPPKAHEV",
                        "VARAGQKGY",
                        "VASEGIKRY",
                        "VHLTPEEK",
                        "VITDKEKAE",
                        "VPKEGKVV",
                        "VQKTIAEN",
                        "VQNGKHFKF",
                        "VSKGTLVQTKGT",
                        "VTTSTRTY",
                        "VYAQKHQQL",
                        "VYGKTSHL",
                        "YADEKTKDV",
                        "YARLHPRAV",
                        "YGSTVSKRV",
                        "YLNKHIQKV",
                        "YPAKAKGTF",
                        "YTKKLNTQ",
                        "YTM(Oxidation)KKVHAL",
                        "YVGDEAQSKR",
                        "YVGDEAQSKRG"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-20T08:25:22.414604503"
    }
}